dashboard/db/migrate/20220315192051_create_programming_classes.rb

```diff
@@ -0,0 +1,21 @@
+class CreateProgrammingClasses < ActiveRecord::Migration[5.2]
+  def change
+    create_table :programming_classes do |t|
+      t.integer :programming_environment_id
```

Comment:
Could you remind me why we need a direct relationship to ProgrammingEnvironment here (instead of inferring it through ProgrammingEnvironmentCategory)?

Reply:
Ah, I see the `validates_uniqueness_of` in ProgrammingExpression now.  Something feels a bit off here but I can't quite articulate it yet.  Maybe we can chat more about the URL requirements some time?  For now, this looks good.

Reply:
That makes sense. I agree with you, but every time I've looked at cleaning this up I've hit the same set of issues (urls and, to a lesser degree, the way we have seeding set up).